### PR TITLE
Anez sandpit

### PR DESCRIPTION
**Pull request by anezm**

### CODE REVIEW

File: src/main/java/java8/challenges/Encoder.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+   String encode(String encode) ;
+
+
+    static Encoder getEncoder() {
+      return input -> {
+         Character[] chars = input.chars()
+                 .mapToObj(ch -> (char) ch)
+                 .toArray(Character[]::new);
+
+         List<Character> charList = Arrays.asList(chars);
```
Comment (subramk) [inline src/main/java/java8/challenges/Encoder.java:27]: can this be fused into the input.chars() as one operation.

File: src/main/java/java8/challenges/Encoder.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+         Set<Map.Entry<Character, Long>> map = getCharacterLongMap(charList).entrySet().stream().collect(Collectors.toSet());
```
Comment (subramk) [inline src/main/java/java8/challenges/Encoder.java:29]: can change method name to  something like characterMapWithCount etc . easier to understand on first go .

File: src/main/java/java8/challenges/Encoder.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import static java.util.stream.Collectors.toList;
+
+@FunctionalInterface
+public interface Encoder {
+   String encode(String encode) ;
+
+
+    static Encoder getEncoder() {
+      return input -> {
+         Character[] chars = input.chars()
```
Comment (subramk) [inline src/main/java/java8/challenges/Encoder.java:23]: overall null check required here . u think ?

File: src/test/java/com/java8/wordsplay/FileReaderTest.java [hunk 4]
```diff
@@ -97,7 +97,7 @@ void shouldReturnWordWithSortedEntries() throws Exception {
                         )
                 )
                .entrySet()
-                .stream()
+               .stream()
                 . sorted( comparing(Entry::getKey)).collect(toList());
 
         wordMap.forEach( entry1 -> System.out.println(entry1.getKey() + " " + entry1.getValue()));
```
Comment (subramk) [inline src/test/java/com/java8/wordsplay/FileReaderTest.java:112]: - WordMap can be empty , then we would have an Optional<Map> coming out ,and we check that before we print ? - Could be a edge case but makes it complete.

File: src/test/java/com/java8/wordsplay/FileReaderTest.java [hunk 5]
```diff
@@ -83,7 +83,7 @@ void shouldReturnWordWithSortedEntries() throws Exception {
 //        - row​: 2
 //                - car​: 1
 //                - die​:
-       String words =  "ant boat row ant car ant die ant boat row ant"  ;
+       String words =  "zan zan zan zan zan zan zan zan ant boat row ant car ant die ant boat row ant zan zan"  ;
 
         String[] wordsArray = words.split(" ");
         List<String> wordList = Arrays.asList(wordsArray);
```
Comment (subramk) [inline src/test/java/com/java8/wordsplay/FileReaderTest.java:97]: objects.nonNull check here before we iterate on the list.

**Review by subramk — CHANGES_REQUESTED** [review]
Ok

File: src/main/java/java8/challenges/Encoder.java (hunk 2 above)
Comment (anezm) [reply src/main/java/java8/challenges/Encoder.java:29]: we can't stream an array thats why converting  into an array list

File: src/test/java/com/java8/wordsplay/FileReaderTest.java (hunk 4 above)
Comment (anezm) [reply src/test/java/com/java8/wordsplay/FileReaderTest.java:112]: Good spot will fix

File: src/test/java/com/java8/wordsplay/FileReaderTest.java (hunk 5 above)
Comment (anezm) [reply src/test/java/com/java8/wordsplay/FileReaderTest.java:97]: Good one will fix

File: src/main/java/java8/challenges/Encoder.java (hunk 2 above)
Comment (subramk) [reply src/main/java/java8/challenges/Encoder.java:29]: Ok , cool

**Review by subramk — APPROVED** [review]
